### ISSUE TITLE
ZnHTTPSTests>>testAmazonAWS has been randomly failing, but the last 4 montly runs have failed, so it is time to look for solutions

### ISSUE THREAD

**Issue by dalehenrich** [GsDevKit/zinc#107]
@kurtkilpela has suggested the following would work:
```
exec
| caCert socket |
caCert := GsSecureSocket setCaCertLocation.
GsFile gciLogClient: caCert.
socket := GsSecureSocket newClient.
socket
  connectTo: 443
  on: 'aws.amazon.com'
  timeoutMs: 50.
"socket setServerNameIndication: 'aws.amazon.com'."
socket secureConnect.
socket
%
```
however, the test framework is a bit more complicated and does not expose the socket so nicely ... so this is a job for @kurtkilpela, himself ...

**Comment by kurtkilpela** [comment]
To clarify a bit, the code in Dale's posting exhibits a similar (perhaps the same) failure as the test case. Inspecting the underlying GsSecureSocket during a failure, it does not appear to set the SNI when ZnClient is used in GemStone.

This information was used to mimic the behavior with a GsSecureSocket.

**Comment by kurtkilpela** [comment]
I have been able to confirm that if I set an SNI when using ZnClient, `https://aws.amazon.com' will load without error.

**Comment by kurtkilpela** [comment]
Resolved by PR #108
